tools/introspection/source/introspection_app.cpp

```diff
@@ -756,54 +756,49 @@ void IntrospectionApp::runIntrospection(const iox::units::Duration updatePeriodM
         {
             bool newPortSampleArrived{false};
             bool newPortThroughputSampleeArrived{false};
-            bool newReceiverPortChangingDataSamplesArrived{false};
+            bool newSubscriberPortChangingDataSamplesArrived{false};
 
-            if (portSubscriber.getChunk(&rawPortSample))
-            {
-                typedPortSample = static_cast<const PortIntrospectionFieldTopic*>(rawPortSample);
-                newPortSampleArrived = true;
-            }
-            if (portThroughputSubscriber.getChunk(&rawPortThroughputSample))
-            {
-                typedPortThroughputSample =
-                    static_cast<const PortThroughputIntrospectionFieldTopic*>(rawPortThroughputSample);
-                newPortThroughputSampleeArrived = true;
-            }
-            if (receiverPortChangingDataSubscriber.getChunk(&rawReceiverPortChangingDataSamples))
-            {
-                typedReceiverPortChangingDataSamples =
-                    static_cast<const ReceiverPortChangingIntrospectionFieldTopic*>(rawReceiverPortChangingDataSamples);
-                newReceiverPortChangingDataSamplesArrived = true;
-            }
+            portSubscriber.take().and_then([&](iox::cxx::optional<iox::popo::Sample<const void>>& maybeSample) {
+                if (maybeSample.has_value())
+                {
+                    typedPortSample = static_cast<const PortIntrospectionFieldTopic*>(maybeSample->get());
```

Comment:
this is a dangling pointer. Just do a `iox::cxx::optional<iox::popo::Sample<const void>> typedPortSample;`. In the `and_then` lambda just do a `typedPortSample = maybeSample;` and replace `newPortSampleArrived` with `typedPortSample.has_value()`.  Don't forget to call `typedPortSample.reset()` at the end of the loop. It seems there were dangling pointer in the old code.